### PR TITLE
module bugfix

### PR DESCRIPTION
**Pull request by Yongqu**
fix debugger break in dynamic attach scenario
When debugger is dynamically attached, we reparse the tree and
regenerate the byte code. During the time we might need to use
the auxilary data. Disable the early release of auxilary data
for now and we'll have a better memory management when we have
full debugger support.
Also fixed a bug when the host pass in nullptr source indicating
download failure.

### CODE REVIEW

**Comment by Yongqu** [comment]
@boingoing can you take a look?

File: lib/Runtime/Language/SourceTextModuleRecord.cpp [hunk 1]
```diff
@@ -91,38 +91,42 @@ namespace Js
         if (sourceText == nullptr)
         {
             Assert(sourceLength == 0);
+            hr = E_FAIL;
             JavascriptError *pError = scriptContext->GetLibrary()->CreateError();
-            JavascriptError::SetErrorMessageProperties(pError, E_FAIL, _u("empty module"), scriptContext);
+            JavascriptError::SetErrorMessageProperties(pError, hr, _u("empty module"), scriptContext);
```
Comment (boingoing) [inline lib/Runtime/Language/SourceTextModuleRecord.cpp]: Should the message say "download failed" or be more descriptive?

File: lib/Runtime/Language/SourceTextModuleRecord.cpp [hunk 2]
```diff
@@ -560,7 +565,6 @@ namespace Js
         }
         Js::AutoDynamicCodeReference dynamicFunctionReference(scriptContext);
         Assert(this == scriptContext->GetLibrary()->GetModuleRecord(srcInfo.moduleID));
-        uint sourceIndex = scriptContext->SaveSourceNoCopy(this->pSourceInfo, static_cast<charcount_t>(this->pSourceInfo->GetCchLength()), /*isCesu8*/ true);
         CompileScriptException se;
         this->rootFunction = scriptContext->GenerateRootFunction(parseTree, sourceIndex, this->parser, this->pSourceInfo->GetParseFlags(), &se, _u("module"));
```
Comment (boingoing) [inline lib/Runtime/Language/SourceTextModuleRecord.cpp]: Where does sourceIndex come from here if not from SaveSourceNoCopy?

File: lib/Runtime/Language/SourceTextModuleRecord.cpp (hunk 1 above)
Comment (Yongqu) [inline lib/Runtime/Language/SourceTextModuleRecord.cpp]: will do.

File: lib/Runtime/Language/SourceTextModuleRecord.cpp (hunk 2 above)
Comment (Yongqu) [inline lib/Runtime/Language/SourceTextModuleRecord.cpp]: It's saved in the modulerecord already. I should have deleted this line :(

**Comment by boingoing** [comment]
LGTM

**Pull request merged**